libtiledbsoma/test/unit_soma_collection.cc

```diff
@@ -49,43 +49,49 @@ TEST_CASE("SOMACollection: basic") {
 }
 
 TEST_CASE("SOMACollection: add SOMASparseNDArray") {
-    TimestampRange ts(0, 2);
-    auto ctx = std::make_shared<SOMAContext>();
-    std::string base_uri = "mem://unit-test-add-sparse-ndarray";
-    std::string sub_uri = "mem://unit-test-add-sparse-ndarray/sub";
-
-    SOMACollection::create(base_uri, ctx, ts);
-    auto index_columns = helper::create_column_index_info(DIM_MAX);
-
-    std::map<std::string, SOMAGroupEntry> expected_map{
-        {"sparse_ndarray", SOMAGroupEntry(sub_uri, "SOMAArray")}};
-
-    auto soma_collection = SOMACollection::open(
-        base_uri, OpenMode::write, ctx, ts);
-    REQUIRE(soma_collection->timestamp() == ts);
-
-    auto soma_sparse = soma_collection->add_new_sparse_ndarray(
-        "sparse_ndarray",
-        sub_uri,
-        URIType::absolute,
-        ctx,
-        "l",
-        ArrowTable(
-            std::move(index_columns.first), std::move(index_columns.second)));
-    REQUIRE(soma_collection->members_map() == expected_map);
-    REQUIRE(soma_sparse->uri() == sub_uri);
-    REQUIRE(soma_sparse->ctx() == ctx);
-    REQUIRE(soma_sparse->type() == "SOMASparseNDArray");
-    REQUIRE(soma_sparse->is_sparse() == true);
-    REQUIRE(soma_sparse->ndim() == 1);
-    REQUIRE(soma_sparse->nnz() == 0);
-    REQUIRE(soma_sparse->timestamp() == ts);
-    soma_sparse->close();
-    soma_collection->close();
-
-    soma_collection = SOMACollection::open(base_uri, OpenMode::read, ctx);
-    REQUIRE(soma_collection->members_map() == expected_map);
-    soma_collection->close();
+    bool use_current_domains[] = {false, true};
```

Comment:
Catch2 has a way to parameterize these options so that they run as separate unit tests. Here is an example from the `SOMAArray` tests: https://github.com/single-cell-data/TileDB-SOMA/blob/main/libtiledbsoma/test/unit_soma_array.cc#L177-L190

Reply:
Commit 3 https://github.com/single-cell-data/TileDB-SOMA/pull/2911/commits/c6f7f730b9400ca55cc5e45c0308cc886d990e10 -- thank you :)